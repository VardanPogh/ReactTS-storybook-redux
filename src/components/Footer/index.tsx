import React from 'react'
import { useHistory, Link } from 'react-router-dom'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Grid from '@material-ui/core/Grid'

import Button from '../Button'

const useStyles = makeStyles(() =>
  createStyles({
    footer: {
      // height: 150,
    },
    logo: {
      width: 200,
    },
    intro: {
      maxWidth: 250,
    },
    links: {
      display: 'block',
      marginBottom: 20,
      textDecoration: 'none',
      fontFamily: 'Poppins, sans-serif',
    },
    footerText: {
      textAlign: 'center',
      backgroundColor: '#014bad',
      color: '#fff',
      padding: 10,
      margin: 0,
    },
  })
)

const Footer: React.FC = () => {
  const classes = useStyles()
  const history = useHistory()
  return (
    <footer className={classes.footer}>
      <Container maxWidth="lg">
        <Grid container spacing={3}>
          <Grid item xs={12} sm>
            <img className={classes.logo} src="img/logo.png" alt="logo" />
          </Grid>

          <Grid item xs={4} sm>
            <Link className={classes.links} to="/faq">
              FAQs
            </Link>
            <Link className={classes.links} to="/about_us">
              About
            </Link>
          </Grid>
          <Grid item xs={4} sm>
            <Link className={classes.links} to="/support">
              Support
            </Link>
            <Link className={classes.links} to="/terms">
              Terms of service
            </Link>
          </Grid>
          <Grid item xs={4} sm>
            <Button
              text="Get Support"
              variant="contained"
              onClick={(): void => history.push('/support')}
            />
          </Grid>
        </Grid>
      </Container>
      <p className={classes.footerText}>
        Copyright Â© BatNav Pty Ltd. All rights reserved.
      </p>
    </footer>
  )
}

export default Footer
