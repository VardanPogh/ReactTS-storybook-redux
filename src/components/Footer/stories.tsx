import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Footer from '.'

export default {
  title: 'Footer',
  component: Footer,
}

export const Default: React.FC = () => (
  <Router>
    <Footer />
  </Router>
)
