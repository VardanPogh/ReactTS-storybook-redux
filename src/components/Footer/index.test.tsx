import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

import Footer from '.'

describe('Footer Test', () => {
  test('Render Footer', () => {
    const { container } = render(
      <Router>
        <Footer />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })
})
