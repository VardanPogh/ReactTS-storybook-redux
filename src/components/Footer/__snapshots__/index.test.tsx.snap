// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test Render Footer 1`] = `
<div>
  <footer
    class="makeStyles-footer-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
        >
          <img
            alt="logo"
            class="makeStyles-logo-2"
            src="img/logo.png"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-true"
        >
          <a
            class="makeStyles-links-4"
            href="/faq"
          >
            FAQs
          </a>
          <a
            class="makeStyles-links-4"
            href="/about_us"
          >
            About
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-true"
        >
          <a
            class="makeStyles-links-4"
            href="/support"
          >
            Support
          </a>
          <a
            class="makeStyles-links-4"
            href="/terms"
          >
            Terms of service
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-6"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Get Support
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <p
      class="makeStyles-footerText-5"
    >
      Copyright Â© BatNav Pty Ltd. All rights reserved.
    </p>
  </footer>
</div>
`;
